src/platformIcon.tsx

```diff
@@ -0,0 +1,167 @@
+  "node-azurefunctions": "azure",
+  "node-connect": "nodejs",
+  "node-express": "express",
+  "node-gcpfunctions": "gcp",
+  "node-koa": "koa",
+  "objective-c": "apple",
+  perl: "perl",
+  php: "php",
+  "php-laravel": "laravel",
+  "php-monolog": "php",
+  "php-symfony2": "symfony",
+  "php-symfony": "symfony",
+  python: "python",
+  "python-aiohttp": "aiohttp",
+  "python-awslambda": "aws",
+  "python-azurefunctions": "azure",
+  "python-bottle": "bottle",
+  "python-celery": "celery",
+  "python-django": "django",
+  "python-falcon": "falcon",
+  "python-flask": "flask",
+  "python-gcpfunctions": "gcp",
+  "python-pylons": "python",
+  "python-pyramid": "pyramid",
+  "python-pythonawslambda": "aws",
+  "python-rq": "redis",
+  "python-sanic": "python",
+  "python-tornado": "tornado",
+  "python-tryton": "tryton",
+  "react-native": "react",
+  ruby: "ruby",
+  "ruby-rack": "ruby",
+  "ruby-rails": "rails",
+  rust: "rust",
+  swift: "swift",
+  // Don't add new platforms down here!
+  // Please add them where they belong alphabetically
+};
+
+function normalizePlatform(platform: string): string {
+  // sentry uses format python-django, but docs uses python.django
+  // this function normalizes that
+  return platform.replace(".", "-");
+}
+
+function getIcon(platform: string): Platform {
+  const normalizedPlatform = normalizePlatform(platform);
+  const icon = PLATFORM_TO_ICON[normalizedPlatform];
+
+  if (icon) {
+    return icon;
+  }
+
+  if (normalizedPlatform.includes("-")) {
+    return getLanguageIcon(normalizedPlatform);
+  }
+
+  return "default";
+}
+
+function getLanguageIcon(platform: string): Platform {
+  const [language] = normalizePlatform(platform).split("-");
+
+  return getIcon(language);
+}
+
+type Platform = keyof typeof PLATFORM_TO_ICON;
+
+type Props = React.HTMLAttributes<HTMLDivElement | HTMLImageElement> & {
+  platform: string;
+  size?: string | number;
+  format?: "sm" | "lg";
+  radius?: number | null;
+  withLanguageIcon?: boolean;
+};
+
+const PlatformIcon = ({
+  platform,
+  size = 20,
+  format = "sm",
+  radius = 3,
+  withLanguageIcon,
+  ...otherProps
+}: Props) => {
+  const icon = getIcon(platform);
+  const iconPath = require(`../${
+    format === "lg" ? "svg_80x80" : "svg"
+  }/${icon}.svg`);
+
+  const languageIcon = getLanguageIcon(platform);
+  const languageIconPath = require(`../svg/${languageIcon}.svg`);
+
+  if (withLanguageIcon && languageIcon !== icon && languageIcon !== "default") {
+    return (
+      <IconContainer {...otherProps}>
+        <Icon src={iconPath} width={size} height={size} radius={radius} />
+        <LanguageIcon src={languageIconPath} />
+      </IconContainer>
+    );
+  }
+
+  return (
+    <Icon
+      src={iconPath}
+      width={size}
+      height={size}
+      radius={radius}
+      {...otherProps}
+    />
+  );
+};
+
+const IconContainer = styled("div")`
+  position: relative;
+`;
+
+const Icon = styled("img")<{ radius?: Props["radius"] }>`
```

Comment:
same here, lets kill emotion, thats huge amounts of bloat for minor inline styles

either that or compile it out for the dist, but dont make me install it

Reply:
Inlining css will make it a bit painful when overriding styles - are we going to be using this on applications without emotion? If we are, then I guess we'll have to inline and mash !important on things, as I don't think the alternatives are any better.

Reply:
well its open source, and you can still override styles without requiring emotion here (e.g. you can pass classname/style/etc)

Reply:
yeah, let's make sure that where-ever we use inline style, to have a prop that can override it

Reply:
Yeah, I am already doing something like this in all places apart from that language baby icon:
https://github.com/getsentry/platformicons/blob/2dff40b71bb5dc4561066f96f7e613363f99bfb1/src/platformIcon.tsx#L132

I can add `languageIconStyles` prop to make overriding of those styles possible.

Fwiw, in sentry, we don't ever override styles of platformicon. In sentry-docs, there are few minor inline styles applied to platformicons (like marginRight). Everything should work flawlessly.